collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -335,9 +337,50 @@ private static boolean isLegalCharacter(char input) {
             (input >= '0' && input <= '9'));
   }
 
-    class Receiver implements JmxScraper.MBeanReceiver {
+    /**
+     * A sample is uniquely identified by its name, labelNames and labelValues
+     */
+    static class SampleKey {
+      String name;
+      List<String> labelNames;
+      List<String> labelValues;
+
+      SampleKey(String name, List<String> labelNames, List<String> labelValues) {
+        this.name = name;
+        this.labelNames = labelNames;
+        this.labelValues = labelValues;
+      }
+
+      static SampleKey of(MetricFamilySamples.Sample sample) {
+        return new SampleKey(sample.name, sample.labelNames, sample.labelValues);
+      }
+
+      @Override
+      public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        SampleKey sampleKey = (SampleKey) o;
+
+        if (name != null ? !name.equals(sampleKey.name) : sampleKey.name != null) return false;
+        if (labelNames != null ? !labelNames.equals(sampleKey.labelNames) : sampleKey.labelNames != null) return false;
+        return labelValues != null ? labelValues.equals(sampleKey.labelValues) : sampleKey.labelValues == null;
+      }
+
+      @Override
+      public int hashCode() {
+        int result = name != null ? name.hashCode() : 0;
+        result = 31 * result + (labelNames != null ? labelNames.hashCode() : 0);
+        result = 31 * result + (labelValues != null ? labelValues.hashCode() : 0);
+        return result;
+      }
+
+    }
+
+    static class Receiver implements JmxScraper.MBeanReceiver {
       Map<String, MetricFamilySamples> metricFamilySamplesMap =
         new HashMap<String, MetricFamilySamples>();
+      Set<SampleKey> uniqueSampleKeys = new HashSet<SampleKey>();
```

Comment:
Not sure about the name `uniqueSampleKeys`. As this is used to check if the sample keys exist, maybe call it `existingSampleKeys`?

Reply:
renamed to `sampleKeys`